### ISSUE TITLE
Improve success consistency of `test_synthDef_isReleased_afterFree`

### ISSUE THREAD

**Issue by capital-G** [supercollider/supercollider#6438]
Currently many test runs fail in CI due to `test_synthDef_isReleased_afterFree` , e.g. in https://github.com/supercollider/supercollider/actions/runs/10649343960/job/29543125921#step:7:3469

```
 	[0/1]	TestNodeProxy_Server:test_synthDef_isReleased_afterFree (2.19s)
		 ! Removing the  NodeProxy source function should remove SynthDef from server
```

Jordan has an idea how to fix this problem

---

              Although adding `server.sync` is wrong, I think my diagnosis might still be correct. 

What we want to do in this case (the test is using `Server.bootSync`) is wait until the server is booted  *and* `ServerBoot` has finished, that way, we get a stable node count.

Here is some code from `Volume` that is called inside of `ServerBoot`.
```supercollider
sendSynthDef {
	fork {
		SynthDef(...).send(server);
		...
		server.sync;
		...
	}
}
```
This boils down to the following example...
```supercollider
fork {
	// boot server
	// start ServerBoot
	// Volume's func
	fork {
		s.sync;
		\FinishedServerBoot.postln;
	};
	
	// the test
	s.sync;
	\DoingTest.postln;
}
```

What we want to see is `\FinishedServerBoot` then `\DoingTest`, but the actual order depends on how the threads are woken up. On my machine, I always get `\DoingTest` then `\FinishedServerBoot`. I don't know if there is a defined or predictable ordering, but this isn't something to be relied upon.

One way to check this would be replacing the `server.sync` in test_synthDef_isReleased_afterFree with `2.wait` (or something), on my machine, this seems to work, but because it's infrequent, I can't say for certain.

_Originally posted by @JordanHendersonMusic in https://github.com/supercollider/supercollider/issues/6433#issuecomment-2321768779_

**Comment by telephon** [comment]
> On my machine, I always get `\DoingTest` then `\FinishedServerBoot`. I

This is correct. In the your example, the order reads like: first wait for DoingTest, then for FinishedServerBoot. The forked routine will run later than the code in which you started it.

If the code should only run once the server has booted, there needs to be a `waitForBoot` or maybe a `s.sync` before the inner fork.

**Comment by telephon** [comment]
I have sometimes run into problems with `s.sync` which I couldn't find the cause for. So I am also not sure here.

**Comment by JordanHendersonMusic** [comment]
> If the code should only run once the server has booted, there needs to be a waitForBoot or maybe a s.sync before the inner fork.

Could you explain this a little more. I'm thinking of the following example.

```
SeverBoot.add({
    SynthDef(\a, ...);
    s.sync;
    Synth(\a)
});
```
This is equivalent to...
```
fork {
	// boot server
	fork {
            SynthDef(...);
            s.sync;  /// SyncA
            Synth(\a);
        };
		
	s.sync; /// SyncB
	\Done
}
```

We want SyncA to run before SyncB, but *I think* this code will always run SyncB first, as the other thread hasn't been switched to yet. 

Alternatively, consider this...

```
SeverBoot.add({
    1.wait;
    SynthDef(\a, ...);
    s.sync;
    Synth(\a)
});

....
fork {
	// boot server
	fork {
           1.wait;
            SynthDef(...);
            s.sync;  /// SyncA
            Synth(\a);
        };
		
	s.sync; /// SyncB
	\Done
}
```
Perhaps `ServerBoot` needs a way to tell `Server.bootSync` than it has finished?

**Comment by JordanHendersonMusic** [comment]
Just to give another example of what I thought might be the cause of error because I explained myself poorly above...
```supercollider
ServerBoot.add({
	1.wait;
	"First".postln;
})

fork {
	s.bootSync;
	"Second".postln;
}

// prints "Second" then "First"

```
